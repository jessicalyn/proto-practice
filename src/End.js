import React, {Component} from 'react';

export default class End extends Component {

  render = () => {
    const {studyGuideQuestions, useStudyGuide, clearStorage} = this.props
    return (
      <div className="question-container">
        { studyGuideQuestions.length === 0 &&
          <div>
            <h2>Congrats, you completed ProtoPractice!</h2>
            <button className="start-over large-button" onClick={clearStorage}>Start Over</button>
          </div>
        }
        { studyGuideQuestions.length >= 1 &&
          <div>
            <h2>Keep Going!</h2>
            <h3>You have {studyGuideQuestions.length} questions in your Study Guide.</h3>
            <button className="study-guide large-button" onClick={useStudyGuide}>Practice Study Guide</button>
            <button className="start-over large-button" onClick={clearStorage}>Start Over</button>
          </div>
        }
      </div>
    )
  }
}

