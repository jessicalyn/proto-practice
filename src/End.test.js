import React from 'react';
import End from './End.js';
import { shallow } from 'enzyme';

const mockStudyGuideQuestions = [{
  answerList: ["push", "includes", "shift"],
  category: "Array Prototype",
  correctAnswer: "shift",
  docs: "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift",
  id: "a72913de-36c6-11e9-b210-d663bd873d93",
  img: "images/shift.png",
  },
  {
  answerList: ["push", "includes", "shift"],
  category: "Array Prototype",
  correctAnswer: "shift",
  docs: "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift",
  id: "a72913de-36c6-11e9-b210-d663bd873d93",
  img: "images/shift.png",
}];
const clearStorageMock = jest.fn();
const useStudyGuideMock = jest.fn();

describe('End', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <End
        studyGuideQuestions={mockStudyGuideQuestions}
        clearStorage={clearStorageMock}
        useStudyGuide={useStudyGuideMock}
      />
    )
  });

  it('should match a snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should fire clear storage method when clicked', () => {
    wrapper.find('.start-over').simulate('click');
    expect(clearStorageMock).toBeCalled();
  });

  it('should fire use study guide method when clicked', () => {
    wrapper.find('.study-guide').simulate('click');
    expect(useStudyGuideMock).toBeCalled();
  });

})