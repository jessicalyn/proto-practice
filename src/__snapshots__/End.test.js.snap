// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`End should match a snapshot with all data passed in 1`] = `
<div
  className="question-container"
>
  <div>
    <h2>
      Keep Going!
    </h2>
    <h3>
      You have 
      2
       questions in your Study Guide.
    </h3>
    <button
      className="study-guide large-button"
      onClick={[MockFunction]}
    >
      Practice Study Guide
    </button>
    <button
      className="start-over large-button"
      onClick={[MockFunction]}
    >
      Start Over
    </button>
  </div>
</div>
`;
